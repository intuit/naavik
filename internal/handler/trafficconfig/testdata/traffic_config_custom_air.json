{
  "apiVersion": "admiral.io/v1",
  "kind": "TrafficConfig",
  "metadata": {
    "generation": 11,
    "labels": {
      "asset": "httpbin",
      "env": "e2e"
    },
    "name": "httpbin",
    "namespace": "admiral"
  },
  "spec": {
    "edgeService": {
      "filters": [
        {
          "name": "filter-3387192"
        },
        {
          "name": "filter--671448791"
        }
      ],
      "routes": [
        {
          "filterSelector": "filter-3387192",
          "inbound": "/health/full",
          "name": "Health Check",
          "outbound": "/health/full",
          "timeout": 2500,
          "workloadEnvSelectors": [
            "e2e"
          ]
        },
        {
          "filterSelector": "filter--671448791",
          "inbound": "/*",
          "name": "v1",
          "outbound": "/*",
          "timeout": 2500,
          "workloadEnvSelectors": [
            "e2e",
            "e2e-air"
          ]
        },
        {
          "filterSelector": "filter--671448791",
          "inbound": "/*",
          "name": "v2",
          "outbound": "/*",
          "timeout": 2500,
          "workloadEnvSelectors": [
            "e2e",
            "e2e-air"
          ]
        }
      ]
    },
    "quotaGroup": {
      "totalQuotaGroups": [
        {
          "cpuLimit": 80,
          "description": "Total capacity of service",
          "memoryLimit": 80,
          "name": "Total Throttling Plan",
          "podLevelThreshold": 100,
          "quotas": [
            {
              "algorithm": "scalable",
              "behaviour": "softThrottle",
              "keyType": "any",
              "maxAmount": 100,
              "name": "Total",
              "rule": "/*",
              "timePeriod": "1s"
            }
          ],
          "regionLevelLimit": true,
          "workloadEnvSelectors": [
            "e2e",
            "e2e-air"
          ]
        }
      ],
      "appQuotaGroups": [
        {
          "name": "sample app plan-2",
          "description": "Some Description",
          "workloadEnvSelectors": [
            "e2e",
            "e2e-air"
          ],
          "associatedApps": [
            "httpbin",
            "some other app"
          ],
          "quotas": [
            {
              "name": "Everything else rule",
              "rule": "/v1*",
              "maxAmount": 15,
              "timePeriod": "1m",
              "keyType": "appIdOrToken",
              "algorithm": "fast",
              "behaviour": "defaultThrottle",
              "method": ""
            }
          ]
        }
      ]
    },
    "workloadEnvs": [
      "e2e",
      "e2e-air"
    ]
  }
}
