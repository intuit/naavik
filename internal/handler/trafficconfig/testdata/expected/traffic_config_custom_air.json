{
  "hosts": [
    "e2e.httpbin.mesh",
    "e2e.httpbin.intuit",
    "e2e-air.httpbin.mesh"
  ],
  "http": [
    {
      "name": "Health Check-e2e",
      "match": [
        {
          "authority": {
            "exact": "e2e.httpbin.mesh"
          },
          "headers": {
            "x-intuit-route-name": {
              "exact": "Health Check"
            }
          }
        }
      ],
      "route": [
        {
          "destination": {
            "host": "e2e.httpbin.mesh"
          }
        }
      ]
    },
    {
      "name": "v1-e2e",
      "match": [
        {
          "authority": {
            "exact": "e2e.httpbin.mesh"
          },
          "headers": {
            "x-intuit-route-name": {
              "exact": "v1"
            }
          }
        }
      ],
      "route": [
        {
          "destination": {
            "host": "e2e.httpbin.mesh"
          }
        }
      ]
    },
    {
      "name": "v1-e2e-air",
      "match": [
        {
          "authority": {
            "exact": "e2e.httpbin.intuit"
          },
          "headers": {
            "x-intuit-route-name": {
              "exact": "v1"
            }
          }
        }
      ],
      "route": [
        {
          "destination": {
            "host": "e2e-air.httpbin.mesh"
          }
        }
      ]
    },
    {
      "name": "v2-e2e",
      "match": [
        {
          "authority": {
            "exact": "e2e.httpbin.mesh"
          },
          "headers": {
            "x-intuit-route-name": {
              "exact": "v2"
            }
          }
        }
      ],
      "route": [
        {
          "destination": {
            "host": "e2e.httpbin.mesh"
          }
        }
      ]
    },
    {
      "name": "v2-e2e-air",
      "match": [
        {
          "authority": {
            "exact": "e2e.httpbin.intuit"
          },
          "headers": {
            "x-intuit-route-name": {
              "exact": "v2"
            }
          }
        }
      ],
      "route": [
        {
          "destination": {
            "host": "e2e-air.httpbin.mesh"
          }
        }
      ]
    },
    {
      "name": "defaultall-e2e",
      "match": [
        {
          "uri": {
            "prefix": "/"
          },
          "authority": {
            "prefix": "e2e.httpbin.mesh"
          }
        }
      ],
      "route": [
        {
          "destination": {
            "host": "e2e.httpbin.mesh"
          }
        }
      ]
    },
    {
      "name": "defaultall-e2e-air-mesh",
      "match": [
        {
          "uri": {
            "prefix": "/"
          },
          "authority": {
            "prefix": "e2e-air.httpbin.mesh"
          }
        }
      ],
      "route": [
        {
          "destination": {
            "host": "e2e-air.httpbin.mesh"
          }
        }
      ]
    },
    {
      "name": "defaultall-e2e-air",
      "match": [
        {
          "uri": {
            "prefix": "/"
          },
          "authority": {
            "prefix": "e2e.httpbin.intuit"
          }
        }
      ],
      "route": [
        {
          "destination": {
            "host": "e2e-air.httpbin.mesh"
          }
        }
      ]
    }
  ]
}
